mysensors/mysensors.py

```diff
@@ -552,6 +558,15 @@ def _check_socket(self, sock=None, timeout=None):
             raise OSError
         return available_socks
 
+    def tcp_check(self):
+        if not self._stop_tcp_check.is_set():
+            t = threading.Timer(10.0, self.tcp_check)    #send heartbeat every 10 seconds until socket is disconnected
```

Comment:
Why do you start a new thread here? You just need to check time and compare that to the last time the tcp check was done. Save the time of the check in an instance variable when the check is done. If time is not enough just return.  Call the method from the run loop.